### DISCUSSION TITLE
Resolving Waiting Close and Unconfirmed Transactions

### DISCUSSION THREAD

**Discussion by jimbrend** [lightningnetwork/lnd#8006] [Q&A] [answered]
### Background

I currently have [this guide](https://community.umbrel.com/t/steps-for-waiting-close-channels-and-unconfirmed-transactions/12937) for Umbrel users, any feedback appreciated

Trying to follow the [LND guide ](https://community.umbrel.com/t/steps-for-waiting-close-channels-and-unconfirmed-transactions/12937)
and prepend Umbrel lncli command, 

I'm confused on which portion should be used and in each circumstance as per the LND doc,

Does bumpfee for CPFP work in most scenarios on channel closures?

and it must be this kind of command?

`sudo ~/umbrel/scripts/app compose lightning exec lnd lncli listchaintxns --start_height <current block> --end_height -1`

Or more something like this?

`sudo ~/umbrel/scripts/app compose lightning exec lnd lncli wallet bumpfee --conf_target 6 --force <txid>:<index point>`

Or can I confirm the recommended command step-by-step you should use to speed up the sat per byte on the closings showing unconfirmed? 

I'd like to explain this easier and separate what is and isn't possible, 

For waiting close and unconfirmed LND closings, should we bump fee or use the --start_height command? 

How can we tell for sure when closing between us and the peer if RBF is enabled and what to use? 

How can we tell if that user is a CLN peer?

Thanks for input

### Your environment

* version of `lnd`: LND 0.16.4-beta
* which operating system (`uname -a` on *Nix): umbrelOS v0.5.5 
* version of `btcd`, `bitcoind`, or other backend: Bitcoin Core 25.0.0
* any other relevant environment details: 
Raspberry Pi 4, (umbrelOS also runs on Debian and Ubuntu) 

### Steps to reproduce

Closing all channels in LND

### Expected behaviour

All funds returning onchain 

### Actual behaviour

Channels are stuck waiting close for longer than 1 week

**Comment by xgfreon** [comment]
Here is the pending's I currently have waiting to close. any of them with Tor address do not have a unconfirmed transaction yet in the pool. The clearnet nodes do have unconfirmed. I want to speed those up as they have been set to 10sat/vb and will never clear. 
    "total_limbo_balance": "2311852",
    "pending_open_channels": [
    ],
    "pending_closing_channels": [
    ],
    "pending_force_closing_channels": [
    ],
    "waiting_close_channels": [
        {
            "channel": {
                "remote_node_pub": "02b09fc8ee937c2fbbe5b82405610ba90da35323923d2f1eeacb1e7836d19f200d",
                "channel_point": "fe2c5c653d891957d65d1eba34a878f67a7bad99102c226368a1a6074b48fabe:0",
                "capacity": "1000000",
                "local_balance": "16176",
                "remote_balance": "980353",
                "local_chan_reserve_sat": "10000",
                "remote_chan_reserve_sat": "10000",
                "initiator": "INITIATOR_LOCAL",
                "commitment_type": "ANCHORS",
                "num_forwarding_packages": "6",
                "chan_status_flags": "ChanStatusLocalDataLoss"
            },
            "limbo_balance": "16176",
            "commitments": {
                "local_txid": "39c46d81b6f69625ec97bd1a0d96d15a5d49b5571d31acc6beb7abbd6afe7431",
                "remote_txid": "f9270fee0b88618d33035d91e9c4830fbc3d8bc26b2dc8a84cfb5e00393c0a17",
                "remote_pending_txid": "",
                "local_commit_fee_sat": "2810",
                "remote_commit_fee_sat": "2810",
                "remote_pending_commit_fee_sat": "0"
            },
            "closing_txid": ""
        },
        {
            "channel": {
                "remote_node_pub": "03449f7a53a92501b0ed80da24d530d525b0d08121a7bb97c370251780b75e11a4",
                "channel_point": "4a07940d0830a78e5b4535c17c89efe3454415e0c5a8288753c552d6066c72ab:1",
                "capacity": "250000",
                "local_balance": "53098",
                "remote_balance": "193545",
                "local_chan_reserve_sat": "2500",
                "remote_chan_reserve_sat": "2500",
                "initiator": "INITIATOR_REMOTE",
                "commitment_type": "ANCHORS",
                "num_forwarding_packages": "6",
                "chan_status_flags": "ChanStatusLocalDataLoss"
            },
            "limbo_balance": "53098",
            "commitments": {
                "local_txid": "ab4ca1a7a304a8e87b02e3a8cdc69b1960c4a04d38b7481f33e22a0785afe882",
                "remote_txid": "9a88c226ed0a2dae58a527185e05f76b2de6c9816354a46881b351865af0fb90",
                "remote_pending_txid": "",
                "local_commit_fee_sat": "2696",
                "remote_commit_fee_sat": "2696",
                "remote_pending_commit_fee_sat": "0"
            },
            "closing_txid": ""
        },
        {
            "channel": {
                "remote_node_pub": "0344e747758b9c1085c6f784fc9ccacc790182d958793dfd07058d35eb7a661a95",
                "channel_point": "48141793b86babe8e1897641494859e53780e1fbedc70430a40d50c54ff67b7d:1",
                "capacity": "1000000",
                "local_balance": "549276",
                "remote_balance": "447253",
                "local_chan_reserve_sat": "10000",
                "remote_chan_reserve_sat": "10000",
                "initiator": "INITIATOR_LOCAL",
                "commitment_type": "ANCHORS",
                "num_forwarding_packages": "31",
                "chan_status_flags": "ChanStatusLocalDataLoss"
            },
            "limbo_balance": "549276",
            "commitments": {
                "local_txid": "3edd4a6c860a04fd76cd916eb23af228410e19031491bfae7beac5f468f3e08f",
                "remote_txid": "c552716f79fc538eac6ec5f17b367e922329d92fda7b41c47dbf4ded39459b0d",
                "remote_pending_txid": "",
                "local_commit_fee_sat": "2810",
                "remote_commit_fee_sat": "2810",
                "remote_pending_commit_fee_sat": "0"
            },
            "closing_txid": ""
        },
        {
            "channel": {
                "remote_node_pub": "034ff2a80dcca440c34973c154792248e0df97c7650dc941c78b1258cc98578934",
                "channel_point": "fe5801b134c3b5c54a8597a630f98bf9ff68418ece0a66b9d4a2022bac598f91:1",
                "capacity": "2000000",
                "local_balance": "982548",
                "remote_balance": "1013981",
                "local_chan_reserve_sat": "20000",
                "remote_chan_reserve_sat": "20000",
                "initiator": "INITIATOR_LOCAL",
                "commitment_type": "ANCHORS",
                "num_forwarding_packages": "38",
                "chan_status_flags": "ChanStatusLocalDataLoss"
            },
            "limbo_balance": "982548",
            "commitments": {
                "local_txid": "ac48f54ab052c916dae67b63a978a94911ccfd4a7d10eddabcad97cd607de567",
                "remote_txid": "9a90628e40ac5d2c9fdfa7d90398cc808d7c5e44703dea3bc20ff9ced434b6b2",
                "remote_pending_txid": "",
                "local_commit_fee_sat": "2810",
                "remote_commit_fee_sat": "2810",
                "remote_pending_commit_fee_sat": "0"
            },
            "closing_txid": ""
        },
        {
            "channel": {
                "remote_node_pub": "03a67551e7609f5af5329ff5c7bd04ea1c30e33cd49eb2815c4b10e3932d14dfc6",
                "channel_point": "18353ba7ef65900a15a4f71f92f3edb630f3f6c7a711c224008d01f85b5cdc42:0",
                "capacity": "1000000",
                "local_balance": "175413",
                "remote_balance": "821230",
                "local_chan_reserve_sat": "10000",
                "remote_chan_reserve_sat": "10000",
                "initiator": "INITIATOR_REMOTE",
                "commitment_type": "ANCHORS",
                "num_forwarding_packages": "10",
                "chan_status_flags": "ChanStatusLocalDataLoss"
            },
            "limbo_balance": "175413",
            "commitments": {
                "local_txid": "08535421e12d1bc31c1439824834a9152a75dc126b5fece43f834c45aed5e27e",
                "remote_txid": "22289400409c0b00535729546ccd5ccf72adbf61a37fc62b2ad56c1d630a2ab8",
                "remote_pending_txid": "",
                "local_commit_fee_sat": "2696",
                "remote_commit_fee_sat": "2696",
                "remote_pending_commit_fee_sat": "0"
            },
            "closing_txid": ""
        },
        {
            "channel": {
                "remote_node_pub": "03b279522ced2ccd9d2c7d1fc500d24f567129fc12b9495a076e30a2e861fffb1d",
                "channel_point": "5cfd6224e4d58763a5cdadba62a17aa33c3dabd39960210202d299b53f364eb5:1",
                "capacity": "500000",
                "local_balance": "471646",
                "remote_balance": "24883",
                "local_chan_reserve_sat": "5000",
                "remote_chan_reserve_sat": "5000",
                "initiator": "INITIATOR_REMOTE",
                "commitment_type": "ANCHORS",
                "num_forwarding_packages": "24",
                "chan_status_flags": "ChanStatusLocalDataLoss"
            },
            "limbo_balance": "471646",
            "commitments": {
                "local_txid": "84efea14923b3854a179c2b9369162a2d70b6fbaa5e4057531ab7329a8f6e320",
                "remote_txid": "061922f915d8c18ed1204ac0522134e03bcbd8ea340aba9381d7c1cd9ba135b7",
                "remote_pending_txid": "",
                "local_commit_fee_sat": "2810",
                "remote_commit_fee_sat": "2810",
                "remote_pending_commit_fee_sat": "0"
            },
            "closing_txid": ""
        },
        {
            "channel": {
                "remote_node_pub": "03fe8461ebc025880b58021c540e0b7782bb2bcdc99da9822f5c6d2184a59b8f69",
                "channel_point": "5c63c335d68789855e5dc34ff567b33791d625675432055b397ac45286bf167b:0",
                "capacity": "1000000",
                "local_balance": "63695",
                "remote_balance": "933030",
                "local_chan_reserve_sat": "10000",
                "remote_chan_reserve_sat": "10000",
                "initiator": "INITIATOR_LOCAL",
                "commitment_type": "STATIC_REMOTE_KEY",
                "num_forwarding_packages": "21",
                "chan_status_flags": "ChanStatusLocalDataLoss"
            },
            "limbo_balance": "63695",
            "commitments": {
                "local_txid": "91df04051c20a824f4cfa1a4d0b296eb70f13d707eb22545f454da6808fdfc3f",
                "remote_txid": "820e0b2d2ae38d29b41c0a75a5483d213809d7c91b85c1f5403c5f270d0790b1",
                "remote_pending_txid": "",
                "local_commit_fee_sat": "3274",
                "remote_commit_fee_sat": "3274",
                "remote_pending_commit_fee_sat": "0"
            },
            "closing_txid": ""

**Comment by xgfreon** [comment]
I  have tried to bump close fee using the following command against the channelpoint of the channel open, it didnt return an error but I have not seen anything happen either. 
 ~/umbrel/scripts/app compose lightning exec lnd lncli wallet bumpclosefee --sat_per_byte 50 4a07940d0830a78e5b4535c17c89efe3454415e0c5a8288753c552d6066c72ab:1

**Comment by alexbosworth** [comment]
You don't want to bump the channel outpoint, you want to bump the anchor outpoint

**Comment by jimbrend** [comment]
Thank you for your reply @alexbosworth, 

Myself and others were having trouble with this, is it true [from reference:](https://docs.lightning.engineering/lightning-network-tools/lnd/unconfirmed-bitcoin-transactions) 

> "You will only be able to make use of this command if it was created as an anchor channel. For other channels, only the counterparty of the party initiating the force close will be able to get the channel confirmed more quickly if they spend their output with a higher fee (CPFP)."

And I couldn't find an example of the anchor outpoint command?

We are not able to use it in this scenario @xgfreon shared above?  What would you recommend?  Your input is really appreciated - want to learn and then help others having issues after switching setups/closures too

**Comment by alexbosworth** [comment, marked as answer]
The anchor output you can see is the 330 small outputs on the pending unconfirmed transaction

Bump fee can spend this outpoint into a child tx and then if that tx is a high fee it pays for the parent transaction to confirm, that's CPFP

**Comment by jimbrend** [comment]
@alexbosworth Forgive me if I'm not yet grasping, 

I grab his anchor outpoint from the mempool, for this tx as an example? `4a07940d0830a78e5b4535c17c89efe3454415e0c5a8288753c552d6066c72ab`
https://mempool.space/tx/4a07940d0830a78e5b4535c17c89efe3454415e0c5a8288753c552d6066c72ab

and it would be 

`lncli wallet bumpclosefee --sat_per_byte 50 < ? >:<index point>`

Where does one grab that ?

tysm

**Comment by alexbosworth** [comment]
The close for that tx is `907f0ece05abcdd0ea9c7b16a3012320b3deec102ad896599e2540ab89cafa32`

The anchors on that are index 0 and 1

You can see it is already being CPFP'ed via the index 1

**Comment by jimbrend** [comment]
:grimacing::zap: I haven't been able to find where you got this from
> `907f0ece05abcdd0ea9c7b16a3012320b3deec102ad896599e2540ab89cafa32` 


@alexbosworth

**Comment by alexbosworth** [comment]
That is the id of the unconfirmed tx that spends the channel outpoint, the commitment transaction

**Comment by xgfreon** [comment]
![image](https://github.com/lightningnetwork/lnd/assets/17713661/342cc9db-01a0-4197-9daf-b39d61580570)
 He got it from clicking on that red arrow @usernameisJim 
These close transactions must not have come from my node, as when i try to bumpclosefee on that transaction I get that it was not from my wallet.

**Comment by jimbrend** [comment]
Yeah my understanding is we can bump but we can't CPFP unless you also own the outputs to the TX...
so they must have CPFP'd already?

**Comment by xgfreon** [comment]
@alexbosworth if you look at another of my pending channels from above, in Lnd-g It says channel closed, however when i goto the mempool and look for a close like
 I do for the other example, I dont see a red arror next to my open channel point. How can I find the close for this channel, or maybe its been unreachable from my node and will keep trying to close on its own?
![image](https://github.com/lightningnetwork/lnd/assets/17713661/11fa6bbd-3a23-4940-b489-9afd182a16b5)

https://mempool.space/tx/fe5801b134c3b5c54a8597a630f98bf9ff68418ece0a66b9d4a2022bac598f91

I have this issue for any of my channels that were TOR only channels. I have tried to send a force close using Chantools but it tells me the address is not reachable, I have only been successful using chantools on clearnode ip's only. The node clearly shows online from amboss or 1ML.  
https://1ml.com/node/034ff2a80dcca440c34973c154792248e0df97c7650dc941c78b1258cc98578934
https://amboss.space/node/034ff2a80dcca440c34973c154792248e0df97c7650dc941c78b1258cc98578934

**Comment by jimbrend** [comment]
@xgfreon maybe you can contact them their twitter and Telegram are linked [their](https://amboss.space/node/034ff2a80dcca440c34973c154792248e0df97c7650dc941c78b1258cc98578934) to initiate the cooperative close?

**Comment by alexbosworth** [comment]
I don't have enough info from what you posted but if you want to close a channel and you still have the channel state you can try to coop close or force close the channel

If a force close gets stuck in the mempool and it's an anchor channel and the parent is above the minimum relay fee you can fee bump the anchor

**Comment by saubyk** [comment]
Moving this issue to a discussion thread, as there is nothing to fix here.

**Comment by ElucGeek** [comment]
I'm in the same situation but my channel is not anchor, it's "commitment_type":  "STATIC_REMOTE_KEY".
What can I do?

**Reply by guggero** [reply]
Unfortunately with the `STATIC_REMOTE_KEY` channel type you cannot do anything, at least not easily... That's the exact reason anchor channels were created in the first place.

If you can communicate with your peer and they are willing to cooperate, you could use the zombie recovery flow described here: https://github.com/lightninglabs/chantools/blob/master/doc/zombierecovery.md

**Reply by ElucGeek** [reply]
That's a pitty. I don't even know why this channel (and maybe other) are in this type and not anchor, are they just too old? How to prevent opening channel with STATIC_REMOTE_KEY in the future?

I contacted the peer already, but the procedure seams extremely complicated. We will see what is possible. Remain an external accelerator service but I don't know which one to use, mempool space one is on waiting list only, I don't know other that are trustfull.

**Reply by guggero** [reply]
Not all implementations supported anchor channels until recently. But if you open new channels between `lnd` and `lnd` they will by default be new anchor channels. So yes, the channels are likely just quite old or with other implementations.

**Reply by jimbrend** [reply]
as @guggero states his zombie tool links you with the missing peer and is from my understanding, 
basically a notification service that imo should be shared more as it is a great development, to know if someone resurfaces their lightning channels,

@nmfretz iirc the new flow you developed, will more easily let @ElucGeek see any channel had he not seen them reliably on the opt-in and friendly UI if he is _not_ not updated?-- albiet, may be unrelated to that lengthy tree of possible outcomes[ here in the diagram? under "Channel recovery scenario"](https://github.com/lightninglabs/chantools)
yay SCB files lol

**Reply by jimbrend** [reply]
afaik @ElucGeek just run pending channels command and hope your peer accepts your close broadcast if they are around right?
resource also at very top of thread: https://community.umbrel.com/t/steps-for-waiting-close-channels-and-unconfirmed-transactions/12937 

perhaps i can still try to iron that out hopefully there is some kind of resolution for this like a remotesweep after identifying if the peer is eclair or something, or something with a user friendly UI :)
